protocol/syncer.go

```diff
@@ -508,82 +355,8 @@ func (s *Syncer) DeletePeer(peerID peer.ID) error {
 	return nil
 }
 
-// findCommonAncestor returns the common ancestor header and fork
-func (s *Syncer) findCommonAncestor(clt proto.V1Client, status *Status) (*types.Header, *types.Header, error) {
-	h := s.blockchain.Header()
-
-	min := uint64(0) // genesis
-	max := h.Number
-
-	targetHeight := status.Number
-
-	if heightNumber := targetHeight; max > heightNumber {
-		max = heightNumber
-	}
-
-	var header *types.Header
-
-	for min <= max {
-		m := uint64(math.Floor(float64(min+max) / 2))
-
-		if m == 0 {
-			// our common ancestor is the genesis
-			genesis, ok := s.blockchain.GetHeaderByNumber(0)
-			if !ok {
-				return nil, nil, ErrLoadLocalGenesisFailed
-			}
-
-			header = genesis
-
-			break
-		}
-
-		found, err := getHeader(clt, &m, nil)
-		if err != nil {
-			return nil, nil, err
-		}
-
-		if found == nil {
-			// peer does not have the m peer, search in lower bounds
-			max = m - 1
-		} else {
-			expectedHeader, ok := s.blockchain.GetHeaderByNumber(m)
-			if !ok {
-				return nil, nil, fmt.Errorf("cannot find the header %d in local chain", m)
-			}
-			if expectedHeader.Hash == found.Hash {
-				header = found
-				min = m + 1
-			} else {
-				if m == 0 {
-					return nil, nil, ErrMismatchGenesis
-				}
-				max = m - 1
-			}
-		}
-	}
-
-	if header == nil {
-		return nil, nil, ErrCommonAncestorNotFound
-	}
-
-	// get the block fork
-	forkNum := header.Number + 1
-	fork, err := getHeader(clt, &forkNum, nil)
-
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to get fork at num %d", header.Number)
-	}
-
-	if fork == nil {
-		return nil, nil, ErrForkNotFound
-	}
-
-	return header, fork, nil
-}
-
 // WatchSyncWithPeer subscribes and adds peer's latest block
-func (s *Syncer) WatchSyncWithPeer(p *SyncPeer, handler func(b *types.Block) bool) {
+func (s *Syncer) WatchSyncWithPeer(p *SyncPeer, newBlockHandler func(b *types.Block) bool, blockTimeout time.Duration) {
 	// purge from the cache of broadcasted blocks all the ones we have written so far
```

Comment:
nitpick: broadcasted is not standard English, "broadcast" is used for past tense :)

Reply:
Lol hahahaha